main.go

```diff
@@ -196,16 +308,28 @@ func main() {
 		}
 	}
 
-	subj := "faas-request"
-	qgroup = "faas"
-
-	log.Println("Wait for ", config.AckWait)
-	sub, err := sc.QueueSubscribe(subj, qgroup, mcb, startOpt, stan.DurableName(durable), stan.MaxInflight(config.MaxInflight), stan.AckWait(config.AckWait))
-	if err != nil {
-		log.Panicln(err)
+	natsQueue := NatsQueue{
+		clusterID: "faas-cluster",
+		clientID:  "faas-worker-" + nats.GetClientID(hostname),
+		natsURL:   "nats://" + config.NatsAddress + ":4222",
+
+		connMutex:      &sync.RWMutex{},
+		maxReconnect:   config.MaxReconnect,
+		reconnectDelay: config.ReconnectDelay,
+		quitCh:         make(chan struct{}),
```

Comment:
Whilst making changes for the other comments, can `struct{}` be exchanged for `bool` or `int`?

Reply:
I strongly disagree with that one. This is common pattern to do signalling channels when you are not interested in the value being sent to the channel to use `struct{}` because it is not allocating any memory for that.